File: php/last2.php
<?php
	function last2($str)
	{
		$count=0;
		$last2=substr($str, -2);
		for ($i=0; $i < strlen($str)-2; $i++) 
		{ 
			if (substr($str, $i,2)==$last2) 
			{
				$count+=1;
			}
		}
		return $count;
	}
echo last2("hixxhi")==1 ? "pass" : "fail"; echo "\n";
echo last2("xaxxaxaxx")==1 ? "pass" : "fail"; echo "\n";
echo last2("axxxaaxx")==2 ? "pass" : "fail"; echo "\n";
echo last2("aabbaabbaabbaa")==3 ? "pass" : "fail"; echo "\n";
?>

File: ruby/last2.rb
def last2(str)
	last2=str.split(//).last(2).join
	count=0
	strlen=str.length
	for i in 0..(strlen-4)
		if str[i..(i+1)]==last2
			count+=1
		end
	end
	return count
end
